### PR TITLE
Fix: Correctly escape newlines in `constructType` function

### LINKED ISSUE #5699: 🐛 BUG: wrangler types generates invalid JS for multi-line strings
### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

3.51.2

### What version of Node are you using?

v21.7.1

### What operating system and version are you using?

Mac Sonoma 14.4.1

### Observed behavior

When using a multi-line string as environment variable, `wrangler types` generates invalid JS.

### Expected behavior

`wrangler types` should generate valid JS using either \` or by inserting \n.

### Steps to reproduce
- Add a multi-line string: https://github.com/Le0Developer/wrangler-types-multiline/blob/master/wrangler.toml#L13
- Run `wrangler types`
- Observe invalid JS: https://github.com/Le0Developer/wrangler-types-multiline/blob/master/worker-configuration.d.ts#L5-L6

### Please provide a link to a minimal reproduction

https://github.com/Le0Developer/wrangler-types-multiline

### Please provide any relevant error logs

_No response_

### PR DESCRIPTION
**Pull request by harugon**
## What this PR solves / how to test

Fixes #5699

This PR fixes the issue with `constructType` function handling multiline strings improperly. It ensures that newlines in strings are correctly escaped, preventing invalid JavaScript code generation. 

## Author has addressed the following

- Tests
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: This change affects type generation only
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: Internal functionality change only

### CODE REVIEW

File: packages/wrangler/src/type-generation.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ export function constructType(
 		if (useRawVal) {
 			return `${typeKey}: ${value};`;
 		}
-		return `${typeKey}: "${value.replace(/"/g, '\\"')}";`;
+		return `${typeKey}: "${value.replace(/"/g, '\\"').replace(/\n/g, '\\n')}";`;
```
Comment (penalosa) [inline packages/wrangler/src/type-generation.ts:147]: I wonder if we'd be able to get away with something simpler here—maybe `JSON.stringify()` of the vlaue?

Comment (harugon) [reply packages/wrangler/src/type-generation.ts]: Thank you for the suggestion! I've updated the constructType function to use JSON.stringify() for escaping string values. Please review the changes and let me know if there are any further improvements needed.

File: packages/wrangler/src/type-generation.ts [hunk 2]
```diff
@@ -144,7 +144,7 @@ export function constructType(
 		if (useRawVal) {
 			return `${typeKey}: ${value};`;
 		}
-		return `${typeKey}: "${value.replace(/"/g, '\\"')}";`;
+		return `${typeKey}: "${JSON.stringify(value)}";`;
```
Comment (penalosa) [inline packages/wrangler/src/type-generation.ts:147]: I don't think this needs wrapped in quotation marks?
```suggestion
		return `${typeKey}: ${JSON.stringify(value)};`;
```

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
After running the tests, I noticed that the original backslash escaping tests were not correct either. So this fix actually fixes that too.

**Pull request merged**